### PR TITLE
Adding Test Plan draft

### PR DESCRIPTION
**Pull request by tybar**
First pull request!

I am not sure what all guidance people will need, so I can make this more granular as needed. I wanted to start out high level first, and drill down as needed.

Also, please let me know if there's some fancier Markdown that would look better. :)

### CODE REVIEW

File: docs/test-plan [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+* If in doubt, write a test. You'll never get yelled at for having one too many tests in your code. :)
+
+### What not to test
+* Not every card needs a test. This includes plain dudes without any card text, as well as relatively simple cards like Tusk.
+
+### How to test
+* Start by identifying the main "happy" path for the code you're writing.
+* Then identify any edge cases that are relevant for the you're writing.
+* Finally, start building up tests. Your tests should test functionality, not implementation (i.e. they should not need to be updated because of internal changes to what code you're writing)
+
```
Comment (tybar) [inline docs/test-plan:22]: Should have put in something to the effect of "make sure to ad-hoc new cards and interactions with the client too."

**Comment by mmeldo** [comment]
Congrats to your first pull request. The plan covers almost all, but I believe in current situation with only me doing engine changes, we do not need such broad plan. What I would like to do now is identify areas that are not properly covered by tests. There are already some issues with `tests` label to identify such areas. And of course to implement those tests afterwards.

**Review by mmeldo — COMMENTED** [review]
Give the `test-plan` file `.md` extension so GitHub knows it should treat it as markdown.

**Comment by tybar** [comment]
> Give the `test-plan` file `.md` extension so GitHub knows it should treat it as markdown.

Done.
